### PR TITLE
ADQ-72 datamart allergy intolerance

### PR DESCRIPTION
**Pull request by jblefkove-va**
https://vasdvp.atlassian.net/browse/ADQ-72

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/AllergyIntoleranceController.java [hunk 1]
```diff
@@ -142,4 +217,75 @@ public OperationOutcome validate(@RequestBody AllergyIntolerance.Bundle bundle)
...
+
+  /**
+   * This class is being used to help organize the code such that all the datamart logic is
+   * contained together. In the future when Mr. Anderson support is dropped, this class can be
+   * eliminated.
+   */
+  private class Datamart {
+
+    AllergyIntoleranceEntity findById(@PathVariable("publicId") String publicId) {
+      MultiValueMap<String, String> publicParameters = Parameters.forIdentity(publicId);
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/AllergyIntoleranceController.java]: public-to-cdw-id lookup can not be done with `gov.va.api.health.dataquery.service.controller.WitnessProtection#toCdwId`

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/AllergyIntoleranceController.java [hunk 2]
```diff
@@ -142,4 +217,75 @@ public OperationOutcome validate(@RequestBody AllergyIntolerance.Bundle bundle)
...
+      witnessProtection.registerAndUpdateReferences(
+          resources,
+          resource ->
+              Stream.concat(
+                  Stream.of(resource.recorder().orElse(null), resource.patient().orElse(null)),
+                  resource.notes().stream().map(n -> n.practitioner().orElse(null))));
+    }
+
+    AllergyIntolerance.Bundle searchByPatient(MultiValueMap<String, String> publicParameters) {
+      MultiValueMap<String, String> cdwParameters =
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/AllergyIntoleranceController.java]: `gov.va.api.health.dataquery.service.controller.WitnessProtection#toCdwId` for icn

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/AllergyIntoleranceController.java [hunk 3]
```diff
@@ -142,4 +217,75 @@ public OperationOutcome validate(@RequestBody AllergyIntolerance.Bundle bundle)
...
+          witnessProtection.replacePublicIdsWithCdwIds(publicParameters);
+      String icn = cdwParameters.getFirst("patient");
+      int page = Integer.parseInt(cdwParameters.getOrDefault("page", asList("1")).get(0));
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/AllergyIntoleranceController.java]: I'm not a real big fan of the Integer parse that is starting to show up in several places. Let's talk about what we could do to keep the simplicity of just passing the parameters object around.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/Parameters.java [hunk 4]
```diff
@@ -11,8 +11,18 @@
 /** Provides utilities for working with MultiValueMap typically used for request parameters. */
 @NoArgsConstructor(staticName = "builder")
 public class Parameters {
+
   private final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
 
+  /** Return first '_count' value or the default. */
+  public static int countOf(@NonNull MultiValueMap<String, String> parameters, int defaultVal) {
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/Parameters.java]: I'd be OK if these or there was a version that assumed a default value.

`int count = Parameters.countOf(params))`

**Comment by jblefkove-va** [comment]
This should be ready now

**Pull request merged**